[TM] Adds new ruin on lavaland with ghost-role spawners.

EgorDinamit opened the pull request:
## About The Pull Request

Adds tech cult base on the lavaland, alongside with Energy Sniper rifle, unique to it.

## Why It's Good For The Game

Ruins are always interesting for miners/explorers.
You will be able to research more weapons, more powerful cybernetic organs and implants if you unlock the Marsian Technology by using one of their items with destructive analyzer(similar to unlocking alien/illegal tech).
Leader of this cult also counts as "holy-role". That means he might be useful as the chaplain on the station if you convince them to help you.

![lavaland_surface_techcult-1](https://user-images.githubusercontent.com/53223414/83000474-770b1e80-a01b-11ea-8acd-b89e19a398a7.png)

## Changelog
:cl:
add: Added Tech Cult base on the lavaland. It contains 5 ghost-roles, new tier4 cybernetic organs, weapons and implants. The base itself is a combination of R&D, Surgery and Chapel.
add: Added hidden tech-node for all of the tech cult technology. To unlock it, you simply put one of their items into the destructive analyzer and use it. Similar to unlocking alien/illegal tech.
/:cl:

EgorDinamit committed:
Main update

EgorDinamit committed:
Main update2

LemonInTheDark commented:
I'm glad you found a place that likes your stuff egor, the drama was painful. 

Update your changelog too

EgorDinamit commented:
I will update it a bit later, I just placed it here so people will see the main code and maybe idea of this ruin.

EgorDinamit committed:
another big update

EgorDinamit committed:
Update techcult.dm

EgorDinamit committed:
up

EgorDinamit committed:
Update tgstation.dme

EgorDinamit committed:
Update techcult.dm

EgorDinamit committed:
EMP-Proof limbs, R&D fixup, etc.

EgorDinamit committed:
TM-Ready

ATH1909 commented:
oh boy another lavaland loot pinata

ATH1909 commented on code/modules/ruins/lavalandruin_code/techcult.dm:
```diff
@@ -0,0 +1,592 @@
+	. = ..()
+	AddComponent(/datum/component/anti_magic, FALSE, FALSE, TRUE, ITEM_SLOT_OCLOTHING)
+
+/obj/item/clothing/head/hooded/techpriest/armor/lead
```
er, this has the exact same typepath as the robes

EgorDinamit replied:
Nope, it's perfectly fine.
"clothing/head", not "clothing/suit"

ATH1909 commented on code/modules/ruins/lavalandruin_code/techcult.dm:
```diff
@@ -0,0 +1,592 @@
+	armor = list("melee" = 45, "bullet" = 10, "laser" = 25, "energy" = 25, "bomb" = 25, "bio" = 0, "rad" = 0, "fire" = 60, "acid" = 60)
+	hoodtype = /obj/item/clothing/head/hooded/techpriest/armor
+
+/obj/item/clothing/head/hooded/techpriest/armor
```
same problem here as below

EgorDinamit replied:
Perfectly fine as above.

ATH1909 commented on code/modules/ruins/lavalandruin_code/techcult.dm:
```diff
@@ -0,0 +1,592 @@
+	icon = 'icons/obj/machines/sleeper.dmi'
+	icon_state = "sleeper"
+	short_desc = "You are a member of the machine cult on Lavaland."
+	flavour_text = "The flesh is weak and humans are fragile. You exist only to research universe and enchance abilities with power of science."
```
```suggestion
	flavour_text = "The flesh is weak and humans are fragile. You exist only to research the universe and enchance your abilities with the power of science."
```

ATH1909 commented on code/modules/ruins/lavalandruin_code/techcult.dm:
```diff
@@ -0,0 +1,592 @@
+	new_spawn.grant_language(/datum/language/machine, TRUE, TRUE, LANGUAGE_MIND) //They have to implant robotic box instead of a tongue to speak this language though.
+
+/datum/outfit/techcult/post_equip(mob/living/carbon/human/H)
+	H.faction |= "Mechanicus"
```
make sure that their armor won't reject them before they receive the mechanicus faction

EgorDinamit replied:
How dare you assume I didn't test it?

Yes, it works.

ATH1909 commented on code/modules/ruins/lavalandruin_code/techcult.dm:
```diff
@@ -0,0 +1,592 @@
+	spell = /obj/effect/proc_holder/spell/aoe_turf/conjure/tech
+	spellname = "holy gift"
+	icon_state ="bookcharge"
+	remarks = list("But will it make me a mech?", "I could've just use the exosuit fabricator, huh?", "All powerful god, send me a honkbot!", "So it is a pocket robotic factory.", "This page is just full of binary code...", "Can it give me a laser arm?", "But can I get combat implants from here?")
```
```suggestion
	remarks = list("But will it make me a mech?", "I could've just used the exosuit fabricator, huh?", "All-powerful god, send me a honkbot!", "So it is a pocket robotics factory.", "This page is just full of binary code...", "Can it give me a laser arm?", "But can I get combat implants from here?")
```

ATH1909 commented on code/modules/ruins/lavalandruin_code/techcult.dm:
```diff
@@ -0,0 +1,592 @@
+	dose_available = TRUE
+	emp_vulnerability = 5
+
+/obj/item/organ/heart/cybernetic/tier4/on_life()
```
shouldn't this already ben inherited from normal cybernetic hearts?

EgorDinamit replied:
This proc is on tier2 and tier 3 only, **as far as I know.**
Additionally, this one is a bit different.

ATH1909 commented on code/modules/ruins/lavalandruin_code/techcult.dm:
```diff
@@ -0,0 +1,592 @@
+		if(RG.total_volume < RG.maximum_volume)
+			RG.add_reagent(reagent_ids[i], 5)
+
+/obj/item/reagent_containers/borghypo/hacked/augment/add_reagent(datum/reagent/reagent)
```
check to see if the parent item already has this stuff

EgorDinamit replied:
Parent has this stuff, but it's a shit-code.
Volume variable is never used so I'm calling this to change it from 30 to 10.

ATH1909 commented on code/modules/ruins/lavalandruin_code/techcult.dm:
```diff
@@ -0,0 +1,592 @@
+/obj/item/organ/cyberimp/arm/surgery/plus/emag_act(mob/user)
+	if(!(locate(/obj/item/reagent_containers/borghypo/hacked/augment) in items_list))
+		to_chat(user, "<span class='notice'>You unlock and hack integrated hypospray located in [src]!</span>") //Oh god oh fuck
+		items_list += new /obj/item/reagent_containers/borghypo/hacked/augment(src)
```
jesus christ what the fuck why

I mean, I guess fulp usually doesn't allow murderboning, and getting this requires raiding/visiting a specific lavaland ruin as a traitor (or someone who has access to an emag, at any rate), so it'll be hard to abuse this TOO badly, but still...

Eh, on second thought, you could just use a sleepypen (or an emagged mediborg) anyway, I guess it's okay.

EgorDinamit replied:
It's quite balanced.
Keep in mind the SLOW regeneration rate and total volume, which is 10 for each toxin.

EgorDinamit committed:
Update code/modules/ruins/lavalandruin_code/techcult.dm

Co-authored-by: ATH1909 <42606352+ATH1909@users.noreply.github.com>

EgorDinamit committed:
Update code/modules/ruins/lavalandruin_code/techcult.dm

Co-authored-by: ATH1909 <42606352+ATH1909@users.noreply.github.com>

EgorDinamit committed:
inhand_icon_state

EgorDinamit committed:
inhand_icon_state 2

EgorDinamit committed:
Update fulp_overwrite_vars.dm

BarrelOfNapkins commented:
Looks cool!

EgorDinamit committed:
Update

Rebalanced armor, edited closet contents, upgraded reviver implant plus, fixed bug with leader's armor.

EgorDinamit committed:
Removes emp-proof augs.

Moved them to another PR.

EgorDinamit committed:
Update fulp_overwrite_vars.dm

EgorDinamit committed:
Update human_defense.dm

EgorDinamit committed:
Le update.

Changes energy sniper rifle weight_class to Normal, instead of Bulky. It also fits on the back now.
Changes Tech Cult specific bible to work on Cyborgs and Augmented people only.

EgorDinamit committed:
Update techcult.dm

BarrelOfNapkins commented:
NO!!! IT MUST STAY OPEN!!

EgorDinamit committed:
Merge pull request #4 from TheSwain/master

fm

EgorDinamit committed:
update
